src/CountIterableCapableTrait.php

```diff
@@ -34,17 +38,13 @@
     protected function _countIterable($iterable)
     {
         $resolve = function ($it) {
-            while (!($it instanceof Countable)) {
-                if (!($it instanceof IteratorAggregate)) {
-                    break;
-                }
-
-                $it = $it->getIterator();
+            try {
+                return $this->_resolveIterator($it, function ($it) {
+                    return $it instanceof Countable;
+                });
+            } catch (RootException $e) {
+                return;
```

Comment:
Why returning nothing (i.e. `null`) when an exception is thrown? I would assume the exception should either bubble up, get wrapped, or at least document that `null` can be returned.

Reply:
Document where?

Reply:
Sorry. I didn't realize this was a closure. I see now that the return value is later checked against certain types.